Tests/UnitTests/EdgeHitProcessorTests.swift

```diff
@@ -0,0 +1,129 @@
+            XCTAssertTrue(success)
+            expectation.fulfill()
+        }
+
+        // verify
+        wait(for: [expectation], timeout: 1)
+        XCTAssertFalse(mockNetworkService?.connectAsyncCalled ?? true) // no network request should have been made
+    }
+
+    /// Tests that when a good hit is processed that a network request is made and the request returns 200
+    func testProcessHitHappy() {
+        // setup
+        let expectation = XCTestExpectation(description: "Callback should be invoked with true signaling this hit should not be retried")
+        guard let expectedUrl = URL(string: "adobe.com") else { XCTFail("Could not create URL"); return }
+        let expectedEvent = Event(name: "Hit Event", type: EventType.identity, source: EventSource.requestIdentity, data: nil)
+        let expectedHeaders = ["testHeaderKey": "testHeaderVal"]
+        let expectedRequest = EdgeRequest(meta: nil, xdm: nil, events: nil)
+        let hit = EdgeHit(url: expectedUrl, request: expectedRequest, headers: expectedHeaders, event: expectedEvent)
+
+        mockNetworkService?.connectAsyncMockReturnConnection = HttpConnection(data: "{}".data(using: .utf8), response: HTTPURLResponse(url: expectedUrl, statusCode: 200, httpVersion: nil, headerFields: nil), error: nil)
+
+        let entity = DataEntity(uniqueIdentifier: "test-uuid", timestamp: Date(), data: try? JSONEncoder().encode(hit))
+
+        // test
+        hitProcessor.processHit(entity: entity) { success in
+            XCTAssertTrue(success)
+            expectation.fulfill()
+        }
+
+        // verify
+        wait(for: [expectation], timeout: 1)
+        XCTAssertTrue(mockNetworkService?.connectAsyncCalled ?? false) // network request should have been made
+        XCTAssertEqual(mockNetworkService?.connectAsyncCalledWithNetworkRequest?.url, expectedUrl) // network request should be made with the url in the hit
+    }
+
+    /// Tests that when the network request fails but has a recoverable error that we will retry the hit and do not invoke the response handler for that hit
+    func testProcessHitRecoverableNetworkError() {
+        // setup
+        let expectation = XCTestExpectation(description: "Callback should be invoked with true signaling this hit should be retried")
+        guard let expectedUrl = URL(string: "adobe.com") else { XCTFail("Could not create URL"); return }
+        let expectedEvent = Event(name: "Hit Event", type: EventType.identity, source: EventSource.requestIdentity, data: nil)
+        let expectedHeaders = ["testHeaderKey": "testHeaderVal"]
+        let expectedRequest = EdgeRequest(meta: nil, xdm: nil, events: nil)
+        let hit = EdgeHit(url: expectedUrl, request: expectedRequest, headers: expectedHeaders, event: expectedEvent)
+
+        let edgeResponse = EdgeResponse(requestId: "test-req-id", handle: nil, errors: [EdgeEventError(eventIndex: 0, message: "test-err", code: "502", namespace: nil)], warnings: nil)
```

Comment:
we should test all recoverable error codes somehow, either in this test or testing isRecoverable

Reply:
On the fence here, we're essentially testing the hardcoded array in the EdgeEventError, I think duplicating this test for all values in that array is a bit overkill. We currently don't have any tests for the EdgeEventError, even if we wrote tests for `isRecoverable` we'd just be asserting the list of error codes against another hard coded list that we need to maintain is correct, not sure there' much value gain there.

Reply:
I think we need to test that the retry logic is correctly executed for the recoverable error codes. If the code changes at any point, the tests should be a safeguard of that.

Reply:
I updated this test to iterate over all the possible known recoverable error codes to cover this case.